### ISSUE TITLE
Completed Quest Icons Always Turn Themselves Back On

### ISSUE THREAD

**Issue by CyleJS** [Questie/Questie#5262]
## Bug description
When unchecking the "Completed" box under the "Quests" section of "Show icons for..." in the "Icons" tab, the Completed Quest markers (gold question marks) disappear from the map as expected. However, upon discovering any new area in any zone, the option will reset and Completed Quest markers will begin showing on the map again. The relevant box remains unchecked, but the icons reappear regardless. Checking and unchecking the box makes them disappear again, but they'll always come back as soon as any new map area is discovered.

This has been happening ever since I started using Questie in August. The problem hasn't changed or gone away going from Hardcore into Season of Discovery.

## Screenshots
![Completed Icons Error](https://github.com/Questie/Questie/assets/60512676/3e9e627c-f8bd-4f09-91cd-b54f1c535aea)

## Questie & WoW version
Questie version 9.1.0
Wow Classic Era version 1.15.0 (52409)

**Comment by Cabro** [comment]
This is first time i see this bug reported. Thank you

**Comment by MyTechnoHunter** [comment]
does this happen with only Questie enabled?

**Comment by CyleJS** [comment]
> does this happen with only Questie enabled?

Yes, it still happens even if Questie is the only enabled addon. Tested it with a new character just now.

**Comment by BreakBB** [comment]
Thanks for looking into this @TorpedoCXC 👍🏻

**Comment by CyleJS** [comment]
@BreakBB @TorpedoCXC the issue is with the icons continually re-enabling themselves on the main map, not the minimap.

**Comment by TorpedoCXC** [comment]
> @BreakBB @TorpedoCXC the issue is with the icons continually re-enabling themselves on the main map, not the minimap.

I will test this, too. Thx for your continued feedback.

**Comment by TorpedoCXC** [comment]
Before changing the Questie setting
![WoWScrnShot_100824_142608](https://github.com/user-attachments/assets/9d466431-f6dd-49a0-89b2-b15da8e391c8)

Changing the settings
![WoWScrnShot_100824_142617](https://github.com/user-attachments/assets/01681460-8791-4b1f-9c8a-13aed703cbd2)

Afterwards
![WoWScrnShot_100824_142619](https://github.com/user-attachments/assets/d5679e52-a013-4b8c-9a71-47d9d4ae7bab)

Discovering new zone
![WoWScrnShot_100824_142715](https://github.com/user-attachments/assets/ee6c2ec1-5243-4d66-ac3c-afdcf438f92d)

Map
![WoWScrnShot_100824_142841](https://github.com/user-attachments/assets/adbb3731-3938-481b-992c-ccfae58ad6b1)

Sorry. I still can not reproduce the bug. @BreakBB 

But from what I see from your first screenshot, your character may be very close to the question mark. If you are close enough, the question mark will be displayed anyway (by Blizzard, not by Questie). May this be the cause for the perceived bug? Or are you certain that also question marks very far away reappear? @CyleJS

**Comment by BreakBB** [comment]
>  If you are close enough, the question mark will be displayed anyway (by Blizzard, not by Questie)

This is only true for the minimap though @TorpedoCXC 

> But since you can't reproduce it, I guess it's just a problem on my end somehow.
It might be some specific setting situation though 🤔 

Can you share a fresh screenshot of your Icons settings @CyleJS ?

**Comment by CyleJS** [comment]
@TorpedoCXC @BreakBB thanks for continuing to look into this. I've done as you requested and confirmed that, for me, the issue is still identical to when I first reported it. I created a new character, turned off every addon except Questie, and took 3 screenshots. I ran quite far away from the quest turn-in area to make sure that proximity to the quest giver wouldn't affect antything.

In Screenshot 1, you'll see that I have icons enabled and the question mark is showing on the main map (expected behavior, totally fine).
For Screenshot 2, I've disabled Completed Quest icons and the icon has rightly disappeared from the map.
For screenshot 3, I changed nothing but ran a few steps forward and discovered a new map area. However, as you can see, the Completed Quest icon reappeared on the map even though it was still disabled in settings.

Thanks for your continued help with this.
![Screenshot 1](https://github.com/user-attachments/assets/af48b66d-90d0-413a-9144-454a21c9a0db)
![Screenshot 2](https://github.com/user-attachments/assets/992231a2-b513-4b8e-bb7c-75a902b72a68)
![Screenshot 3](https://github.com/user-attachments/assets/ddd37b06-a8e1-4fc8-bd74-1a682127ae9f)

**Comment by TorpedoCXC** [comment]
Thx @CyleJS 

I am out of ideas at the moment, but will give it another thought tomorrow. Looking at your screenshots, there seem to be a bug somewhere indeed. At least for you(r installation).

**Comment by BreakBB** [comment]
@CyleJS thanks a lot for taking the time to reproduce the issue again and providing these screenshots. That surely helps!

@TorpedoCXC Looking at the settings, my best guess is, that this issue is related to the "Hide Icons in Unexplored Areas" setting. Let me know, if you want to look into this again 👍🏻
